userspace/libsinsp/parsers.cpp

```diff
@@ -4499,39 +4499,28 @@ void sinsp_parser::parse_setgid_exit(sinsp_evt *evt)
 	}
 }
 
-static std::string convert_to_string(Json::ValueType other) {
-	std::string value_type("Not a valid json type");
-	switch(other) {
-	case Json::ValueType::nullValue:
-		return "nullValue";
-	case Json::ValueType::intValue:
-		return "intValue";
-	case Json::ValueType::uintValue:
-		return "uintValue";
-	case Json::ValueType::realValue:
-		return "realValue";
-	case Json::ValueType::stringValue:
-		return "stringValue";
-	case Json::ValueType::booleanValue:
-		return "booleanValue";
-	case Json::ValueType::arrayValue:
-		return "arrayValue";
-	case Json::ValueType::objectValue:
-		return "objectValue";
-	default:
-		break;
+namespace
+{
+	std::string generate_error_message(const Json::Value& value, const char* field) {
+		std::string val_as_string = value.isConvertibleTo(Json::stringValue) ? value.asString().c_str() : "value not convertible to string";
+		std::string err_msg = "Unable to convert json value '" + val_as_string + "' for the field: '" + field +"'";
+
+		return std::move(err_msg);
 	}
-	return value_type;
 }
 
-bool sinsp_parser::check_is_convertible(const Json::Value& value, Json::ValueType other, std::string field, bool log_message)
+bool sinsp_parser::check_json_val_is_convertible(const Json::Value& value, Json::ValueType other, const char* field, bool log_message)
 {
-	if(!value.isConvertibleTo(other)) {
-		std::string value_as_string = value.isConvertibleTo(Json::stringValue) ? value.asString() : "value not convertible to string";
-		std::string err_msg = "Unable to convert json value '" + value_as_string + "' of type '" + convert_to_string(value.type()) + "' into type '" + convert_to_string(other) + "' for the field: '" + field +"'";
+	if(!value.isNull() && !value.isConvertibleTo(other)) {
+		std::string err_msg;
+		
 		if(log_message) {
+			err_msg = generate_error_message(value, field);
 			SINSP_WARNING("%s",err_msg.c_str());
 		} else {
+			if(g_logger.get_severity() >= sinsp_logger::SEV_DEBUG) {
+				err_msg = generate_error_message(value, field);
+			}
 			SINSP_DEBUG("%s",err_msg.c_str());
```

Comment:
move this inside the previous `if` block.

Reply:
Done.